pyccel/codegen/printing/ccode.py

```diff
@@ -899,6 +905,9 @@ def _print_Return(self, expr):
         args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
         if expr.stmt:
             code += self._print(expr.stmt)+'\n'
+        if len(self._allocs) > 0:
+            code += self.free_allocs(self._allocs)
+            code += '\n'
```

Comment:
This still feels like it belongs in the `SemanticParser` to me. Could identical code be written in the semantic step if a `Deallocate` class was created?
Then `_print_Deallocate` for fortran would just be:
```
return ''
```
?